packages/notification-center/src/store/notifications-provider.context.tsx

```diff
@@ -1,195 +1,129 @@
+    if (!notificationsPages) {
+      return;
     }
 
-    setNotifications(Object.assign({}, notifications));
-  }
+    const messageIds = notificationsPages.pages.reduce<string[]>((acc, paginatedResponse) => {
+      const pageMessageIds = paginatedResponse.data.filter((message) => !message.read).map((message) => message._id);
 
-  async function refetch(storeId = 'default_store') {
-    setFetching(true);
+      return [...acc, ...pageMessageIds];
+    }, []);
 
-    if (refetchTimeout.get(storeId)) {
-      clearTimeout(refetchTimeout.get(storeId));
-      setRefetchTimeout(refetchTimeout.set(storeId, null));
+    if (messageIds.length > 0) {
+      markNotificationsAs({ messageId: messageIds, seen: true, read: true });
     }
+  }, [markNotificationsAs, notificationsPages]);
 
-    setRefetchTimeout(
-      refetchTimeout.set(
-        storeId,
-        setTimeout(async () => {
-          await fetchPage(0, true, storeId);
-        }, 250)
-      )
-    );
-  }
-
-  function getStoreQuery(storeId: string) {
-    return stores?.find((store) => store.storeId === storeId)?.query || {};
-  }
-
-  async function markAsSeen(
-    messageId?: string,
-    readExist?: boolean,
-    messages?: IMessage | IMessage[],
-    storeId = 'default_store'
-  ) {
-    if (messageId) {
-      await api.markMessageAsSeen(messageId);
+  const markAllNotificationsAsSeen = useCallback(() => {
+    if (!notificationsPages) {
+      return;
     }
 
-    const notificationsToMark = getNotificationsToMark(messages, notifications, storeId);
+    const messageIds = notificationsPages.pages.reduce<string[]>((acc, paginatedResponse) => {
+      const pageMessagesIds = paginatedResponse.data
+        .filter((message) => !message.seen && !message.read)
+        .map((message) => message._id);
 
-    if (notificationsToMark.length) {
-      const notificationsToUpdate = filterReadNotifications(readExist, notificationsToMark);
+      return [...acc, ...pageMessagesIds];
+    }, []);
 
-      const messagesIdsToMark = notificationsToUpdate.map((notification) => notification._id);
-      await api.markMessageAs(messagesIdsToMark, { seen: true });
+    if (messageIds.length > 0) {
+      markNotificationsAs({ messageId: messageIds, seen: true, read: false });
     }
-  }
-
-  function onWidgetClose() {
-    resetPageState();
-  }
-
-  function onTabChange(storeId = 'default_store') {
-    setPage(page.set(storeId, 0));
-  }
-
-  function resetPageState() {
-    setPage(new Map<string, number>([['default_store', 0]]));
-  }
-
-  return (
-    <NotificationsContext.Provider
-      value={{
-        notifications,
-        fetchNextPage,
-        hasNextPage,
-        fetching,
-        markAsRead,
-        updateAction,
-        refetch,
-        markAsSeen,
-        onWidgetClose,
-        onTabChange,
-        markAllAsRead,
-      }}
-    >
-      {children}
-    </NotificationsContext.Provider>
+  }, [markNotificationsAs, notificationsPages]);
```

Comment:
mark all notifications as seen, choose the ones that are not seen and not read from the list of fetched notifications